src/Components/WebAssembly/WebAssembly/src/Services/DefaultWebAssemblyJSRuntime.cs

```diff
@@ -72,5 +73,32 @@ public static void BeginInvokeDotNet(string callId, string assemblyNameOrDotNetO
                 DotNetDispatcher.BeginInvokeDotNet(Instance, state.callInfo, state.argsJson);
             });
         }
+
+        /// <summary>
+        /// Invoked via Mono's JS interop mechanism (invoke_method)
+        /// 
+        /// Notifies .NET of an array that's available for transfer from JS to .NET
+        /// 
+        /// Ideally that byte array would be transferred directly as a parameter on this
+        /// call, however that's not currently possible due to: <INSERT_BUG_HERE></INSERT_BUG_HERE>
+        /// </summary>
+        /// <param name="id">Id of the byte array</param>
+        public static void NotifyByteArrayAvailable(int id)
+        {
+            var data = Instance.InvokeUnmarshalled<byte[]>("Blazor._internal.retrieveByteArray");
+
+            DotNetDispatcher.ReceiveByteArray(Instance, id, data);
+
+            // Is the above technique acceptable or do we have to queue the call as below?
+            // The below approach doesn't work as the call to receive the byte array is executed
+            // AFTER we receive the main begin invoke call (which tries to deserialize the byte array
+            // we haven't yet processed here).
+            //WebAssemblyCallQueue.Schedule((id, data), static state =>
+            //{
+            //    // This is not expected to throw, as it takes care of converting any unhandled user code
+            //    // exceptions into a failure on the Task that was returned when calling InvokeAsync.
+            //    DotNetDispatcher.ReceiveByteArray(Instance, state.id, state.data);
+            //});
```

Comment:
```suggestion
```

Any objections?

Reply:
This technique looks ideal to me. If we later hear back from the Mono people that there's a more direct way to supply the array as a JS-to-.NET parameter then we could switch to that, but otherwise, what you've done here looks perfect.

I'm not aware of any reason to schedule the call in a queue, and doing so would create a bunch of additional things to worry about (e.g., concurrent executions, `byteArrayBeingTransferred` getting overwritten, etc).